java/vector/src/main/java/org/apache/arrow/vector/complex/BaseRepeatedValueVector.java

```diff
@@ -97,11 +98,13 @@ public boolean allocateNewSafe() {
     return dataAlloc;
   }
 
-  protected void allocateOffsetBuffer(final long size) {
+  protected void allocateOffsetBuffer(final long size, boolean init) {
     final int curSize = (int) size;
     offsetBuffer = allocator.buffer(curSize);
     offsetBuffer.readerIndex(0);
-    offsetAllocationSizeInBytes = curSize;
+    if (!init) {
+      offsetAllocationSizeInBytes = curSize;
+    }
```

Comment:
This is to avoid changing `offsetAllocationSizeInBytes` to the initial 4 byte size. Otherwise, `reallocOffsetBuffer` will take it and cause some tests failed.